homeassistant/components/arcam_fmj/media_player.py

```diff
@@ -72,20 +84,12 @@ def __init__(
         """Initialize device."""
         self._state = state
         self._device_name = device_name
-        self._name = f"{device_name} - Zone: {state.zn}"
         self._uuid = uuid
-        self._support = (
-            SUPPORT_SELECT_SOURCE
-            | SUPPORT_PLAY_MEDIA
-            | SUPPORT_BROWSE_MEDIA
-            | SUPPORT_VOLUME_SET
-            | SUPPORT_VOLUME_MUTE
-            | SUPPORT_VOLUME_STEP
-            | SUPPORT_TURN_OFF
-            | SUPPORT_TURN_ON
-        )
         if state.zn == 1:
-            self._support |= SUPPORT_SELECT_SOUND_MODE
+            self._attr_supported_features |= SUPPORT_SELECT_SOUND_MODE
```

Comment:
You can't modify this variable here since you set it up as a class variable. It might be modifying the class instance rather than the object instance. At least it's unclear to me. Move the init of supported features to init()